pkg/workceptor/workceptor.go

```diff
@@ -678,25 +688,91 @@ func (w *Workceptor) unitStatusForCFR(unitID string) (map[string]interface{}, er
 	return retMap, nil
 }
 
+func (w *Workceptor) attemptRemoteAction(remoteNodeID, remoteUnitID, remoteAction string, unit *workUnit) error {
+	timeout := time.After(300 * time.Second)
+	retryInterval := 500 * time.Millisecond
+	shouldTimeout := func() bool {
+		select {
+		case <-timeout:
```

Comment:
The global timeout might be better treated as a `Context` that a caller could pass in.